File: TeamCode/src/main/java/net/gearmaniacs/teamcode/motors/Lift.java
package net.gearmaniacs.teamcode.motors;

import com.qualcomm.robotcore.hardware.DcMotor;
import com.qualcomm.robotcore.hardware.DcMotorEx;
import com.qualcomm.robotcore.hardware.HardwareMap;

import net.gearmaniacs.teamcode.utils.IHardware;
import org.jetbrains.annotations.NotNull;

public final class Lift implements IHardware {

    private DcMotorEx left;
    private DcMotorEx right;

    @Override
    public void init(@NotNull HardwareMap hardwareMap) {
        HardwareMap.DeviceMapping<DcMotor> dcMotors = hardwareMap.dcMotor;
        left = (DcMotorEx) dcMotors.get("lift_left");
        right = (DcMotorEx) dcMotors.get("lift_right");

        left.setMode(DcMotor.RunMode.STOP_AND_RESET_ENCODER);
        right.setMode(DcMotor.RunMode.STOP_AND_RESET_ENCODER);

        left.setTargetPosition(0);
        right.setTargetPosition(0);

        left.setMode(DcMotor.RunMode.RUN_TO_POSITION);
        right.setMode(DcMotor.RunMode.RUN_TO_POSITION);
    }

    @Override
    public final void start() {
    }

    @Override
    public final void stop() {
    }

    @NotNull
    public final DcMotorEx getLeft() {
        return left;
    }

    @NotNull
    public final DcMotorEx getRight() {
        return right;
    }
}

File: TeamCode/src/main/java/net/gearmaniacs/teamcode/hardware/motors/Lift.kt
package net.gearmaniacs.teamcode.hardware.motors

import com.qualcomm.robotcore.hardware.DcMotor
import com.qualcomm.robotcore.hardware.DcMotorEx
import com.qualcomm.robotcore.hardware.DcMotorSimple
import com.qualcomm.robotcore.hardware.HardwareMap
import net.gearmaniacs.teamcode.utils.IHardware
import net.gearmaniacs.teamcode.utils.getDevice

class Lift : IHardware {
    
    lateinit var left: DcMotorEx
        private set
    lateinit var right: DcMotorEx
        private set
    
    override fun init(hardwareMap: HardwareMap) {
        left = hardwareMap.getDevice("lift_left")
        right = hardwareMap.getDevice("lift_right")

        with(left) {
            direction = DcMotorSimple.Direction.REVERSE
            mode = DcMotor.RunMode.STOP_AND_RESET_ENCODER
            targetPosition = 0
            mode = DcMotor.RunMode.RUN_TO_POSITION
        }

        with(right) {
            direction = DcMotorSimple.Direction.FORWARD
            mode = DcMotor.RunMode.STOP_AND_RESET_ENCODER
            targetPosition = 0
            mode = DcMotor.RunMode.RUN_TO_POSITION
        }
    }

    fun setTargetPositionAll(targetPosition: Int) {
        left.targetPosition = targetPosition
        right.targetPosition = targetPosition
    }

    fun setPowerAll(power: Double) {
        left.power = power
        right.power = power
    }
}
